### PR TITLE
patch api/articles/id

### PR DESCRIPTION
**Pull request by PENbDM**

### CODE REVIEW

File: __tests__/topics.test.js [hunk 1]
```diff
@@ -152,7 +152,58 @@ describe('/api/articles/:article_id/comments', () => {
...
+      const existingArticleId = 1;
+
+      const res = await request(app)
+        .patch(`/api/articles/${existingArticleId}`)
+        .send(updateData);
+
+      expect(res.statusCode).toBe(200);
+      expect(res.body).toEqual(
+        expect.objectContaining({
+          article: expect.objectContaining({
```
Comment (roisinoreilly) [inline __tests__/topics.test.js:171]: You know the specific article and all it's properties so I would specify that here instead of expecting any

File: __tests__/topics.test.js [hunk 2]
```diff
@@ -152,7 +152,58 @@ describe('/api/articles/:article_id/comments', () => {
...
+            article_id: existingArticleId,
+            votes: expect.any(Number),
+          }),
+        })
+      );
+    });
+
+    test('400: when missing inc_votes or invalid article_id', async () => {
```
Comment (roisinoreilly) [inline __tests__/topics.test.js:179]: Great edge case

File: __tests__/topics.test.js [hunk 3]
```diff
@@ -152,7 +152,58 @@ describe('/api/articles/:article_id/comments', () => {
...
+      const updateData = {
+      };
 
+      const res = await request(app)
+        .patch('/api/articles/invalidId')
```
Comment (roisinoreilly) [inline __tests__/topics.test.js:184]: I think the invalid id and missing properties in the patch object should be two separate tests

File: controllers/ArticlesController.js [hunk 4]
```diff
@@ -49,20 +49,10 @@ const postCommentForArticleById = (req, res, next) => {
...
-      next(err);
-    });
-};
+
 
 
 
 module.exports ={
-    getArticles,getAllArticles,getArticlesByIdAndComments,postCommentForArticleById
+    getArticles,getAllArticles,getArticlesByIdAndComments,postCommentForArticleById,patchArticlesByID
```
Comment (roisinoreilly) [inline controllers/ArticlesController.js:57]: Where is the patchArticlesByID controller? It's not in this pull request

File: models/ArtcilesModel.js [hunk 5]
```diff
@@ -51,7 +51,21 @@ const fetchCommentCount = (articleId) => {
...
+  const updateArticleVotes = (article_id, inc_votes) => {
+    return db.query(
+      'UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *',
+      [inc_votes, article_id]
+    )
+      .then((result) => {
+        return result.rows[0];
+      })
+      .catch((err) => {
+        throw err;
```
Comment (roisinoreilly) [inline models/ArtcilesModel.js:65]: I wouldn't recommend catching in the models or throwing errors, your controller should be able to catch any errors from invoking the model and pass them to your error handling middleware. Throwing an error isn't handling an error

**Review by roisinoreilly — COMMENTED** [review]
I've left some comments in your code about a test and where you're handling your errors. There isn't a controller anywhere for this ticket, I'm not sure if you forgot to push it up. Can you please push it and send us another PR :)
